### PR TITLE
search and filter

### PR DESCRIPTION
**Pull request by demands14**
## Related Tickets
- [#51781](https://edu-redmine.sun-asterisk.vn/issues/51781)

## WHAT (optional)
- Search movie by its title , description, filter by genres

## HOW
- create controller find by scope, and ajax to view

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by demands14** [comment]
Ready

File: app/controllers/static_pages_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class StaticPagesController < ApplicationController
   def home
-    @movies = Movie.active.limitation
+    @pagy, @movies = pagy Movie.search params[:search]
+    @genres = Genre.all.asc_genre_name
```
Comment (huongnt-2545) [inline app/controllers/static_pages_controller.rb]:
```suggestion
    @genres = Genre.asc_genre_name
```

File: app/views/shared/_movie.html.erb [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
     <div class="product__item">
       <% if movie.image.attached? %>
-        <div class="anime__details__pic" style="background-image: url(<%= url_for movie.image %>);">
-          <div class="comment"><i class="fa fa-comments"></i></div>
-          <div class="view"><i class="fa fa-eye"></i></div>
+          <div class="product__item__pic" style="background-image: url(<%= url_for movie.image %>);">
+        <% else %>
+          <div class="product__item__pic" style="background-image: url(<%= movie.thumbnail %>);">
+        <% end %>
+            <div class="ep1"><%= movie.status %></div>
```
Comment (huongnt-2545) [inline app/views/shared/_movie.html.erb]: sai indent

File: app/views/shared/_movie.html.erb [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+            <div class="comment"><i class="fa fa-comments"></i></div>
+            <div class="view"><i class="fa fa-eye"></i></div>
+          </div>
+        <div class="product__item__text">
+          <ul>
+            <li><%= movie.genre.name %></li>
```
Comment (huongnt-2545) [inline app/views/shared/_movie.html.erb]: dùng delegate

**Comment by demands14** [comment]
Ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**